### PR TITLE
[Improvement](variant) limit the column size on tablet schema (#27399)

### PR DESCRIPTION
**Pull request by eldenmoon**
## Proposed changes
1. limit the column count to default 2048
2. fix get_inverted_index return nullptr when variant's unique id is -1, using it's parent unique id instead
3. avoid add same path subcolumn duplicately in tablet schema
4. make extracted column unique id -1
Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

File: be/src/vec/common/schema_util.cpp [hunk 1]
```diff
@@ -310,7 +310,7 @@ void update_least_common_schema(const std::vector<TabletSchemaSPtr>& schemas,
         TabletColumn common_column;
         // const std::string& column_name = variant_col_name + "." + tuple_paths[i].get_path();
         get_column_by_type(tuple_types[i], tuple_paths[i].get_path(), common_column,
-                           ExtraInfo {.unique_id = variant_col_unique_id,
+                           ExtraInfo {.unique_id = -1,
                                       .parent_unique_id = variant_col_unique_id,
                                       .path_info = tuple_paths[i]});
         common_schema->append_column(common_column);
```
Comment (csun5285) [inline be/src/vec/common/schema_util.cpp:316]: common_schema->append_column(common_column, TabletSchema::ColumnType::VARIANT)

Comment (csun5285) [inline be/src/vec/common/schema_util.cpp:316]: "TabletSchema::ColumnType::VARIANT" means that the appended column is managed using the "_field_path_to_index."

File: regression-test/suites/variant_p0/column_size_limit.groovy [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        CREATE TABLE IF NOT EXISTS ${table_name} (
+            k bigint,
+            v variant
+        )
+        DUPLICATE KEY(`k`)
+        DISTRIBUTED BY HASH(k) BUCKETS 1
+        properties("replication_num" = "1", "disable_auto_compaction" = "false");
+    """
+    try {
+        sql """insert into ${table_name} values (1, '{"a" : 1, "b" : 2, "c" : 3, "d" : 4, "e" : 5}')"""
```
Comment (csun5285) [inline regression-test/suites/variant_p0/column_size_limit.groovy:42]: Execute this SQL again to confirm that the same column won't be added.

Comment (csun5285) [reply regression-test/suites/variant_p0/column_size_limit.groovy:42]: column size is 5 + 1 = 6 ?

File: be/src/vec/common/schema_util.cpp (hunk 1 above)
Comment (csun5285) [reply be/src/vec/common/schema_util.cpp:316]: "TabletSchema::ColumnType::VARIANT" means that the appended column is managed using the "_field_path_to_index."

**Comment by eldenmoon** [comment]
run buildall

File: be/src/vec/common/schema_util.cpp (hunk 1 above)
Comment (eldenmoon) [reply be/src/vec/common/schema_util.cpp]: done

**Review by csun5285 — APPROVED** [review]
LGTM

**Comment by eldenmoon** [comment]
run buildall

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**